x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -153,6 +185,33 @@ export const createAgentPolicyHandler: FleetRequestHandler<
   }
 };
 
+async function createPackagePolicy(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  agentPolicy: AgentPolicy,
+  packageToInstall: string,
+  options: { spaceId: string; user: AuthenticatedUser | undefined }
+) {
+  // If needed, retrieve package information and build a new package policy for the package
+  // NOTE: we ignore failures in attempting to create package policy, since agent policy might have been created
+  // successfully
+  const newPackagePolicy = await packagePolicyService
+    .buildPackagePolicyFromPackage(soClient, packageToInstall)
+    .catch(() => undefined);
+
+  if (!newPackagePolicy) return;
```

Comment:
Can you explain this error condition a bit more? I'm not sure I'm following why we'd expect this to ever throw an error.

Also shouldn't we throw an error of some sort if `buildPackagePolicyFromPackage` returns `undefined`? The only cases I see where that should be possible is where the package isn't installed or the default output doesn't exist. In both cases, I'd expect us unable to continue safely.

Reply:
I see thanks for the explanation. IMO we should throw an error any time the application doesn't do what the user told it to do rather than have a silent failure. It'd probably be best to rollback/delete the agent policy if it the associated package policies couldn't be created. If needed, we could address this as a **immediate** follow up PR.

Reply:
this is done